### ISSUE TITLE
Issue using backticks when no tty is available

### ISSUE THREAD

**Issue by ebmeierj** [xeger/backticks#12]
When I run a docker-compose command (which utilizes backticks) prefixed with 'nohup' I receive the following error. Changing the following lines in module `Backticks::Command.capture` from `streams << STDIN if interactive?` to `streams << STDIN if STDIN.tty? && interactive?` seems to resolve the error.

```
Starting workspace_mysql-schema_1
Traceback (most recent call last):
  File "<string>", line 3, in <module>
  File "compose/cli/main.py", line 88, in main
  File "compose/cli/main.py", line 140, in perform_command
  File "compose/cli/main.py", line 735, in run
  File "compose/cli/main.py", line 1027, in run_one_off_container
  File "site-packages/dockerpty/pty.py", line 334, in start
  File "site-packages/dockerpty/pty.py", line 373, in _hijack_tty
  File "site-packages/dockerpty/io.py", line 378, in flush
OSError: [Errno 5] Input/output error
docker-compose returned -1
```

Reproduction:
* Create a foo.yml docker-compose file with the following contents:
```
version: "2.0"
services:
  hello-world:
    image: "hello-world"
```
* Run `nohup ruby -r backticks -r docker/compose -e 'Docker::Compose::Session.new(Backticks::Runner.new(interactive: true), dir: Dir.pwd, file: "foo.yml").run("hello-world")' && cat nohup.txt`

**Comment by xeger** [comment]
Strange; this doesn't reproduce for me. What version of Docker Compose are you using?

```
docker-compose --version
docker-compose version 1.14.0, build c7bdf9e
```

```
 nohup ruby -r backticks -r docker/compose -e 'Docker::Compose::Session.new(Backticks::Runner.new(interactive: true), dir: Dir.pwd, file: "foo.yml").run("hello-world")'; cat nohup.out

appending output to nohup.out
Creating network "backticks_default" with the default driver
Pulling hello-world (hello-world:latest)...
latest: Pulling from library/hello-world
b04784fba78d: Pull complete
Digest: sha256:f3b3b28a45160805bb16542c9531888519430e9e6d6ffc09d72261b0d26ff74f
Status: Downloaded newer image for hello-world:latest

Hello from Docker!
This message shows that your installation appears to be working correctly.
...
```

**Comment by xeger** [comment]
Reproduced under Linux. I'll adopt your fix since there's no better way to tell whether it's safe to read from STDIN. As an unfortunate side effect, it won't be possible to "script" interactive backticks processes unless the calling process also uses a PTY to send input to backticks. Hopefully this won't be too common.

STDIN is a _very_ odd duck when it's been redirected from /dev/null:

```
tony@trebuchet:~$ nohup ruby -e 'puts STDIN.tty?' ; cat nohup.out
false

tony@trebuchet:~$ nohup ruby -e 'puts STDIN.closed?' ; cat nohup.out
false

tony@trebuchet:~$ nohup ruby -e 'puts STDIN.eof?' ; cat nohup.out
-e:1:in `eof?': Bad file descriptor @ io_fillbuf - fd:0 <STDIN> (Errno::EBADF)
	from -e:1:in `<main>'
```

**Comment by xeger** [comment]
I published backticks v1.0.1 with your fix; thanks for reporting.
